imports/plugins/core/inventory/server/no-meteor/utils/getVariantInventoryNotAvailableToSellQuantity.js

```diff
@@ -0,0 +1,33 @@
+/**
+ *
+ * @method getVariantInventoryNotAvailableToSellQuantity
+ * @summary Get the number of product variants that are currently reserved in an order.
+ * This function can take any variant object.
+ * @param {Object} variant - A product variant object.
+ * @param {Object} collections - Raw mongo collections.
+ * @return {Promise<number>} Reserved variant quantity.
+ */
+export default async function getVariantInventoryNotAvailableToSellQuantity(variant, collections) {
+  // Find orders that are new or processing
+  const orders = await collections.Orders.find({
+    "workflow.status": { $in: ["new", "coreOrderWorkflow/processing"] }
+  }).toArray();
+
+  const reservedQuantity = orders.reduce((sum, order) => {
```

Comment:
@aldeed 

Two questions here:

1) I use a lot of `reduce`s in here. Is this the right way to go about it, or is there a better way to filter down the  different levels arrays?

2) We discussed using the `shipping` or `item` status, instead of the order status, to determine what counts and doesn't count as a `not available` item. I've gone through the different workflows, and here are all the workflows for `order`, `shipping`, and `item`, at different steps. The `inventoryInStock` is decremented at the order step of `click approve`, even though the workflow isn't updated at that point. What do you think is the best status to use for this? @spencern what do you think?

![notes](https://user-images.githubusercontent.com/4482263/50193222-952f0400-02ea-11e9-92af-cb6ba41c905d.png)

Reply:
@aldeed I think this should take care of what you mentioned above: https://github.com/reactioncommerce/reaction/pull/4859/commits/9c37bf6d07562ab9fe2716a8d21b7cd2ae0328dd

One other addition could be to add another status, `coreOrderItemWorkflow/removedFromInventoryQuantity`, and do `item.variantId === variant._id && item.workflow.workflow.includes("coreItemWorkflow/removedFromInventoryAvailableToSell") && !item.workflow.workflow.includes("coreItemWorkflow/removedFromInventoryQuantity")`, is that overkill though?